src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/tasklet/AbstractCatalogBackupRestoreTasklet.java

```diff
@@ -0,0 +1,342 @@
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.backuprestore.tasklet;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.logging.Logger;
+
+import org.apache.commons.io.FileUtils;
+import org.geoserver.backuprestore.Backup;
+import org.geoserver.backuprestore.BackupRestoreItem;
+import org.geoserver.backuprestore.utils.BackupUtils;
+import org.geoserver.config.ServiceInfo;
+import org.geoserver.config.util.XStreamPersisterFactory;
+import org.geoserver.config.util.XStreamServiceLoader;
+import org.geoserver.gwc.config.GWCConfig;
+import org.geoserver.gwc.config.GWCConfigPersister;
+import org.geoserver.gwc.config.GeoserverXMLResourceProvider;
+import org.geoserver.platform.GeoServerExtensions;
+import org.geoserver.platform.GeoServerResourceLoader;
+import org.geoserver.platform.resource.Files;
+import org.geoserver.platform.resource.Paths;
+import org.geoserver.platform.resource.Resource;
+import org.geoserver.platform.resource.Resource.Type;
+import org.geoserver.platform.resource.ResourceStore;
+import org.geoserver.platform.resource.Resources;
+import org.geoserver.platform.resource.Resources.AnyFilter;
+import org.geoserver.util.Filter;
+import org.geotools.util.logging.Logging;
+import org.springframework.batch.core.JobExecution;
+import org.springframework.batch.core.StepContribution;
+import org.springframework.batch.core.scope.context.ChunkContext;
+import org.springframework.batch.core.step.tasklet.Tasklet;
+import org.springframework.batch.repeat.RepeatStatus;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.util.Assert;
+
+/**
+ * TODO
+ * 
+ * @author Alessio Fabiani, GeoSolutions
+ *
+ */
+@SuppressWarnings("rawtypes")
+public abstract class AbstractCatalogBackupRestoreTasklet<T> extends BackupRestoreItem
+        implements Tasklet, InitializingBean {
+
+    protected static Logger LOGGER = Logging.getLogger(AbstractCatalogBackupRestoreTasklet.class);
+
+    /*
+     * 
+     */
+    protected static Map<String, Filter<Resource>> resources = new HashMap<String, Filter<Resource>>();
+
+    /*
+     * 
+     */
+    static {
+        resources.put("demo", AnyFilter.INSTANCE);
+        resources.put("images", AnyFilter.INSTANCE);
+        resources.put("logs", new Filter<Resource>() {
+
+            @Override
+            public boolean accept(Resource res) {
+                if (res.name().endsWith(".properties")) {
+                    return true;
+                }
+                return false;
+            }
+
+        });
+        resources.put("palettes", AnyFilter.INSTANCE);
+        resources.put("plugIns", AnyFilter.INSTANCE);
+        resources.put("styles", new Filter<Resource>() {
```

Comment:
For the main style file, you can use StyleInfo.getFileName, it's going to be a css or a ysld one if the style were edited via rest or via the main style editor. For CSS edited with the custom editor, there is no way to discover them, but you can still just look for the sld file and see if there is a css sidecar one. To locate images one has to roll a style visitor collecting external graphic locations (in the in-memory Style object you should find the absolute paths, they should be getting expanded during parsing)